util/movement-types/src/block.rs

```diff
@@ -184,6 +184,16 @@ impl BlockCommitment {
 	}
 }
 
+impl fmt::Display for BlockCommitment {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		write!(
+			f,
+			"BlockCommitment {{ height: {}, block_id: {}, commitment: {} }}",
```

Comment:
It's better to use the struct helper API of `fmt::Formatter`. It can do pretty formatting when the right mode is specified by the format.